src/chocolatey/GetChocolatey.cs

```diff
@@ -41,8 +41,27 @@ public static class Lets
     {
         public static GetChocolatey GetChocolatey()
         {
+            add_assembly_resolver();
             return new GetChocolatey();
         }
+
+        private static ResolveEventHandler _handler = null;
+
+        private static void add_assembly_resolver()
+        {
+            _handler = (sender, args) =>
+            {
+                var requestedAssembly = new AssemblyName(args.Name);
+                if (requestedAssembly.get_public_key_token().is_equal_to(ApplicationParameters.OfficialChocolateyPublicKey))
+                {
+                     return typeof(Lets).Assembly;

```

Comment:
@RichiCoder1 It's checking the public key token of the requested assembly - meaning that when choco is pulling an item is has merged in, the keys will match. So it will use the internal instead of an external conflicting Assembly. Does this answer your question?